### ISSUE TITLE
Docs: Page does not update when prefers-color-scheme changes

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#18189]
### Docs page(s)

_all of them_

### What documentation issue do you want to solve?

The ESLint docs currently default to the browser's [`prefers-color-scheme`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) setting for light/dark mode by default. Opening or refreshing a new `eslint.org` with no previous `localStorage` will change the page to the preferred light/dark theme.

But, if the browser's `prefers-color-scheme` changes while the page is already open and the user hasn't selected a color scheme override, nothing happens. You have to refresh the page or set the override yourself to see a new light/dark theme.

https://github.com/eslint/eslint/assets/3335181/a2463bad-fc32-4a60-8805-b8b7e3934f54

IMO this is a bug. If the browser indicates a new light/dark mode is preferred, the page should switch to that. Cases where this is a necessary accessibility feature for users include:

* hardware and/or lighting conditions changing while the page is open
* passing control/sight of the browser to a different user
* web developers wanting to quickly change what's being tested

### What do you think is the correct solution?

If the user hasn't selected a color scheme override, I'd like the ESLint docs to update the visual theme when the browser's `prefers-color-scheme` changes.

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

I've even accidentally set a color scheme override because I wanted the page to adhere to my browser/system preference and not have to refresh. But that triggered #18190 and de-synced my page from the system.

Right now, the `<html>` element's `data-theme` is initialized in an inline script:

https://github.com/eslint/eslint/blob/dbf831e31f8eea0bc94df96cd33255579324b66e/docs/src/_includes/layouts/base.html#L77-L81

...while updates are set in a `'DOMContentLoaded'` event:

https://github.com/eslint/eslint/blob/4d1ed22dede531108c8a7899d513f64f0662c135/docs/src/assets/js/themes.js#L11

We can listen for the preferred theme changing with [`window.matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia). From https://stackoverflow.com/questions/59621784/how-to-detect-prefers-color-scheme-change-in-javascript:

```js
function activateDarkMode() {
  // ...
}

const darkModePreference = window.matchMedia("(prefers-color-scheme: dark)");

darkModePreference.addEventListener("change", e => e.matches && activateDarkMode());
```

**Comment by Rec0iL99** [comment]
👍 I could reproduce.

ping @eslint/eslint-team

**Comment by Tanujkanti4441** [comment]
I too could reproduce this in windows, i am up for this fix.

**Comment by nzakas** [comment]
If there's an easy way to fix this, I'd say go for it.
